test/repository_test.rb

```diff
@@ -115,6 +115,28 @@
           UserRepository.create(user1)
           user1.id.must_equal id
         end
+
+        describe 'when entity is not persisted' do
+          before do
+            @unpersisted_user = User.new(name: 'M', age: '23')
+          end
+          it 'assigns and persists created_at attribute' do
+            result = UserRepository.create(@unpersisted_user)
+            result.created_at.must_equal result.created_at
+          end
+        end
+
+        describe 'when entity is already persisted' do
+          before do
+            user              = User.new(name: 'D', age: '25')
+            @user             = UserRepository.create(user)
+            @persisted_user   = UserRepository.persist(@user)
```

Comment:
firstly, why do you use `create` then `persist` here? Should you only use 1 instead?

that is

```
before do
  @persisted_user = UserRepository.persist(User.new(name: 'D', age: '25'))
end

after do
  UserRepository.delete(@persisted_user)
end
```